src/arrayMethodJoin.js

```diff
@@ -4,8 +4,32 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let newSeparator = separator;
+    const newArray = [];
+    let string = '';
+
+    if (!separator) {
+      newSeparator = `${separator}`;
+    }
```

Comment:
Why do you need this check? You have already set a default value for your separator parameter above `separator = ','`

Reply:
Well, I`ve done it at first, but  if i don`t do this,  test with null will ignore null as separator. 
(test(`Should work with null`, () => {
  expect(source.join2(null))
    .toBe('0null1null2null3');
});)